builder/builder-source-archive.c

```diff
@@ -409,6 +410,22 @@ unzip (GFile   *dir,
   return res;
 }
 
+static gboolean
+unrpm (GFile   *dir,
+       const char *rpm_path,
+       GError **error)
+{
+  gboolean res;
+  const gchar *argv[] = { "sh", "-c", NULL, NULL };
+  char *unrpm_cmdline = g_strdup_printf("rpm2cpio %s | cpio -i -d", rpm_path);
+
+  argv[2] = unrpm_cmdline;
+  res = flatpak_spawnv (dir, NULL, error, argv);
+  g_free(unrpm_cmdline);
+
+  return res;
+}
+
```

Comment:
Checking the architecture sounds like a very good thing, however this probably means using librpm (or libhif ?) to do that. Regarding the prefix the rpm ends up in, after the rpm is unpacked, its content has to be make install'ed to /app (ie we need a dummy Makefile with an install: target doing the copying), so the prefix it's unpacked to should not matter too much (and no such check is done for tarballs at the moment).

NB: Maybe I'm using this the wrong way, but what I'm doing is in my manifest is:
"modules": [
        {
            "name": "skype",
            "no-autogen": true,
            "sources" : [
                {
                    "type": "archive",
                    "url": "https://download.skype.com/linux/skypeforlinux-64-alpha.rpm",
                    "sha256": "996e24a59da9a05941f57447dc051f816cc60aaf630135db2dab41bdc5eb7fc0"
                },
                {
                    "type": "file",
                    "path": "skype-Makefile",
                    "dest-filename": "Makefile"
                }
            ],
            "post-install": [
            ]
        }
    ]

and skype-Makefile then has

all:
        true

install:
        cp -ra * /app